### PR TITLE
🐛  useLocalStorage no longer returns a new setValue function every time setValue is called

### PR DESCRIPTION
**Pull request by jaalru**
## Description

Instead of using `storedValue` as a dependency within the `setValue` callback in `createStorageHook.ts` we are getting the current stored value from `setStoredValue` directly. This stops the `setValue` callback from unnecessarily being recreated every time the storedValue changes.

## Related Issue

https://github.com/antonioru/beautiful-react-hooks/issues/395

## Motivation and Context

I was getting an infinite loop when `setValue` was used as a dependency on a useEffect where the effect was calling `setValue`.

https://github.com/antonioru/beautiful-react-hooks/issues/395

## How Has This Been Tested?

Added additional unit tests.

### CODE REVIEW

File: .github/workflows/branch-tests.yml [hunk 1]
```diff
@@ -15,7 +15,8 @@ jobs:
 
     strategy:
       matrix:
-        node-version: 18
+        node-version: [ 18 ]
+        os: [ubuntu-latest]
```
Comment (antonioru) [inline .github/workflows/branch-tests.yml:19]: what's the reason for this change? 
I'm not against it to be honest but perhaps it's worth having a different issue and PR for such a change?

File: src/factory/createStorageHook.ts [hunk 2]
```diff
@@ -54,11 +54,16 @@ const createStorageHook = (type: 'session' | 'local') => {
...
+    const setValue: SetValue<TValue> = useCallback(
+      (value: TValue) => {
+        setStoredValue((current: TValue) => {
+          const valueToStore = value instanceof Function ? value(current) : value
+          safelySetStorage(JSON.stringify(valueToStore))
+          return valueToStore
+        })
+      },
+      [safelySetStorage]
+    )
```
Comment (antonioru) [inline src/factory/createStorageHook.ts:66]: Right, well done here! 👍🏻

**Review by antonioru — CHANGES_REQUESTED** [review]
Hi @jaalru ,

Thank you for your contribution and apologies for the delay in my response. 
Your PR takes useLocalStorage to the next level, thanks 

I have a small request concerning the GitHub action workflow. 
Could you take a moment to review it?

Thank you,

File: .github/workflows/branch-tests.yml (hunk 1 above)
Comment (jaalru) [reply .github/workflows/branch-tests.yml:19]: It can be removed. The pipelines were breaking on my fork and this was reported as a syntax error. 🚧

Comment (jaalru) [reply .github/workflows/branch-tests.yml]: removed.

Comment (antonioru) [reply .github/workflows/branch-tests.yml]: Thanks for removing it, I'll check why it gave you that error and in case I'll open an issue ;) 
thank you @jaalru :)
